examples/step-66/step-66.cc

```diff
@@ -0,0 +1,1156 @@
+  // way as in step-59. We update the ghost values and use the function
+  // integrate_difference from the VectorTools namespace. In the end we gather
+  // all computations from all MPI ranks and return the norm.
+  template <int dim>
+  double GelfandProblem<dim>::compute_solution_norm() const
+  {
+    solution.update_ghost_values();
+
+    Vector<float> norm_per_cell(triangulation.n_active_cells());
+
+    VectorTools::integrate_difference(mapping,
+                                      dof_handler,
+                                      solution,
+                                      Functions::ZeroFunction<dim>(),
+                                      norm_per_cell,
+                                      QGauss<dim>(fe.degree + 2),
+                                      VectorTools::H1_seminorm);
+
+    return VectorTools::compute_global_error(triangulation,
+                                             norm_per_cell,
+                                             VectorTools::H1_seminorm);
+  }
+
+
+
+  // @sect4{GelfandProblem::output_results}
+
+  // The graphical output files in vtu format together with an pvtu master file
+  // we generate in the same way as in step-37. Note, that we write also the
+  // distribution of the triangulation into the output file as it was done in
+  // step-40. So no further comments are necessary.
+  template <int dim>
+  void GelfandProblem<dim>::output_results(const unsigned int cycle) const
+  {
+    if (triangulation.n_global_active_cells() > 1e6)
+      return;
+
+    solution.update_ghost_values();
+
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler(dof_handler);
+    data_out.add_data_vector(solution, "solution");
+
+    Vector<float> subdomain(triangulation.n_active_cells());
+    for (unsigned int i = 0; i < subdomain.size(); ++i)
+      {
+        subdomain(i) = triangulation.locally_owned_subdomain();
+      }
+    data_out.add_data_vector(subdomain, "subdomain");
+
+    data_out.build_patches(fe.degree);
+    std::ofstream output(
+      "solution-" + Utilities::to_string(cycle, 2) + "." +
+      Utilities::to_string(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD),
+                           4) +
+      ".vtu");
+    data_out.write_vtu(output);
+
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+      {
+        std::vector<std::string> filenames;
+        for (unsigned int i = 0;
+             i < Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD);
+             ++i)
+          {
+            filenames.emplace_back("solution-" +
+                                   Utilities::to_string(cycle, 2) + "." +
+                                   Utilities::to_string(i, 4) + ".vtu");
+          }
+        std::ofstream master_output("solution-" +
+                                    Utilities::to_string(cycle, 2) + ".pvtu");
+        data_out.write_pvtu_record(master_output, filenames);
+      }
+  }
```

Comment:
Do you want to use `DataOut::write_vtu_with_pvtu_record`?

Reply:
Sure, thats the function we want to use. Unfortunately, when writing this code, this functions was not yet available.